pkg/apis/core/v1/storagecluster.go

```diff
@@ -611,6 +613,21 @@ const (
 	ClusterOperationTimeout ClusterConditionStatus = "Timeout"
 )
 
+// AnnotationKind is the k8s object type of Portworx components
+// it will be part of the keys to locate a specific Portworx component
+type AnnotationKind string
```

Comment:
Use enum so we can extend later, enum for both AnnotationKind and Component name

Reply:
IMO there is no need to use real "enum" here and it's probably an overkill to add all the map and int values here. 
this strong type of string already has a consistent format compare to other strong type strings in this file already.

Reply:
string type enum is ok, try to encapsulate the isValid function. 
func (t AnnotationKind) IsValid() error {
}